learnEcl/2700-function.md

```diff
@@ -2,3 +2,149 @@
 title: Function
 slug: function
 ---
+
+# FUNCTION
+
+A Function is a set of statements that take input, does some specific computation and produces an output or returns a result. The result could be a value or a dataset.
+
+Notes
+
+* For a function to be called/used from outside of the current module, EXPORT is required.
+* A non-BWR (Builder Window Runnable) file must have an EXPORTed attribute that matches the filename, if not "Error: Definition must contain EXPORT or SHARED" is generated.
+* The EXPORTed attribute must be the last exported attribute in the file. It is perfectly legal to have other (local or SHARED) attributes preceding the final EXPORTed attribute.
+
+## Syntax
+
+<pre>
+<EclCode code="EXPORT [return_data_type] function_name (data_type arg [, data_type arg]+) := FUNCTION
+
+    ecl_code
+    RETURN return_value;
+
+END;">
+</EclCode>
+</pre>
+
+| _Value_ | _Definition_ |
+| :- | :- |
+| EXPORT | Optional. |
+| return_data_type | Optional (compiler can infer it from return_value). If returning a dataset, the data type is DATASET(record_definition). |
+| function_name | The name by which the function will be invoked. |
+| data_type | The argument’s data type. If passing a dataset, the data type is DATASET(record_definition). |
+| ecl_code | Whatever code is needed to build return_value. Conversely, if the code does not contribute to return_value then it is ignored. Attributes defined here are scoped to the function |
+| RETURN | Required |
+| return_value | The result of the function. |
+| END | Required. |
+
+**Example**
+
+<pre>
+<EclCode
+id="FunctionExp_1"
+tryMe="FunctionExp_1"
+code="/*Function Example:*/
+
+/*
+Function Example:
+*/
+
+EXPORT MyFunc (STRING Val) := FUNCTION
+
+   Result := 'Hello ' + Val + ' , welcome to this function';
+   RETURN Result;
+
+END;
+
+// Using MyFunc
+Res := MyFunc('Jonny');
+OUTPUT(Res, NAMED('res'));
+
+OUTPUT(MyFunc('Sunny'), NAMED('Sunny'));
+
+"></EclCode>
+</pre>
+
+## Outputs in Functions - Using WHEN
+
+OUTPUT can be used to return multiple values from a function. 
```

Comment:
OUTPUT() is used to create files or put values into the result section of a workunit.  It does not 'return' anything (it is an action).